src/interfaces/swig/typemaps_utilities.i

```diff
@@ -0,0 +1,74 @@
+/**
```

Comment:
maybe we should add a license here, as well as you as author

Reply:
oh there is an issue here: swig is gpl licensed. So we cannot simply copy paste code into our codebase as it would turn our code into GPL as well....needs to be re-written
On the other hand, I wonder whether someone cares ;)

Reply:
@karlnapf Swig has this clause in their website
```
When SWIG is used as it is distributed by the SWIG developers, its output is not governed by SWIG's license (including the GPL). SWIG's output contains code from three sources:

code generated by SWIG, which is not governed by copyright;
code copied from the SWIG library which is permissively licensed to be redistributed without restriction;
code derived from the user's input, which may be governed by the license of the code supplied by the user.
```
The second source they mention seems to cover our case?

Reply:
Nope, I don't think so, nvm :"D

Reply:
@karlnapf Alright, I think this clears us
```
The SWIG library and examples, under the Lib and Examples top level 
directories, are distributed under the following terms:

  You may copy, modify, distribute, and make derivative works based on
  this software, in source code or object code form, without
  restriction. If you distribute the software to others, you may do
  so according to the terms of your choice. This software is offered as
  is, without warranty of any kind.
```
These macros were taken from files under Lib directory.

Reply:
ok great! thanks for sorting it out!